lldb/source/Core/DynamicLoader.cpp

```diff
@@ -195,20 +196,40 @@ ModuleSP DynamicLoader::LoadBinaryWithUUIDAndAddress(
   Target &target = process->GetTarget();
   Status error;
 
+  StreamString prog_str;
+  if (!name.empty()) {
+    prog_str << name.str() << " ";
+  }
+  if (uuid.IsValid())
+    prog_str << uuid.GetAsString();
+  if (value_is_offset == 0 && value != LLDB_INVALID_ADDRESS) {
+    prog_str << "at 0x";
+    prog_str.PutHex64(value);
+  }
+
   if (!uuid.IsValid() && !value_is_offset) {
+    Progress progress_memread("Reading load commands from memory",
```

Comment:
I would change this string to "Reading object file from memory" instead of "Reading load commands from memory". This function is currently only used by Mach-O core files and the GDB remote stub, but this could be used for loading ELF files from memory as well. 

Maybe this Progress dialog should go into `ReadUnnamedMemoryModule` so anyone that calls this function gets progress events when loading an object file from memory?

Reply:
Good suggestion.  I think I'll conditionalize this progress with a `if (!process->GetCoreFile())` -- reading a binary out of a local corefile will be very fast.  Over a live gdb-remote connection, not so much.